File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/uri/MatchNewActivityUriHandler.java
package com.jdroid.android.sample.ui.uri;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import androidx.annotation.NonNull;

import com.jdroid.android.application.AbstractApplication;
import com.jdroid.android.uri.AbstractUriHandler;

public class MatchNewActivityUriHandler extends AbstractUriHandler<MatchNewActivity> {
	@Override
	public Boolean matches(Uri uri) {
		return true;
	}

	@Override
	public Intent createMainIntent(@NonNull Activity activity, Uri uri) {
		return new Intent(activity, AbstractApplication.get().getHomeActivityClass());
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/uri/MatchNewActivityUriHandler.kt
package com.jdroid.android.sample.ui.uri

import android.app.Activity
import android.content.Intent
import android.net.Uri

import com.jdroid.android.application.AbstractApplication
import com.jdroid.android.uri.AbstractUriHandler

class MatchNewActivityUriHandler : AbstractUriHandler<MatchNewActivity>() {
    override fun matches(uri: Uri): Boolean {
        return true
    }

    override fun createMainIntent(activity: Activity, uri: Uri): Intent? {
        return Intent(activity, AbstractApplication.get().homeActivityClass)
    }
}
